File: medical-clinic-app/src/main/java/com/medicalclinicapp/medicalclinicapp/models/AppointmentHematology.java
package com.medicalclinicapp.medicalclinicapp.models;


import com.fasterxml.jackson.annotation.JsonIgnore;
import com.medicalclinicapp.medicalclinicapp.security.models.Cardiolog;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import javax.persistence.*;
import java.util.Date;

@Entity
@Getter
@Setter
@ToString
@RequiredArgsConstructor
@Table(name="appointmentHematology")
public class AppointmentHematology {
    @OneToOne(cascade = CascadeType.ALL)
    @JsonIgnore
    @JoinColumn(name = "cardiolog_cnp", referencedColumnName = "cnp")
    private Cardiolog cardiolog;

    @OneToOne(mappedBy = "appointmentHematology")
    private Hematology hematology;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    //Analize de sange
    private Date dataAppointment;
    private Number hourAppointment;
    private Number minAppointment;
    private Boolean colesterol_seric_total;
    private Boolean hdl_colesterol;
    private Boolean ldl_colesterol;
    private Boolean trigliceride_serice;
    private Boolean glicemie;
    private Boolean tgo;
    private Boolean tgp;
    private Boolean uree_serica;
    private Boolean creatina_serica;
    private Boolean potasiu_seric;
    private Boolean magneziu_seric;
    private Boolean acid_uric;
    private Boolean calciu_ionic_seric;
    private Boolean calciu_seric_total;
    private Boolean inr_cu_interpretare;
    private Boolean hemoleucograma_completa;
    private Boolean t3;
    private Boolean t4;
    private Boolean tsh;
}

File: medical-clinic-app-front/src/app/interfaces/appointmentHematology.ts
import { Patient } from "./patient";

export class AppointmentsHematology {
    
    dataAppointmentHematology: string;
    hourAppointmentHematology: string;

    //Analize de sange
    colesterol_seric_total: boolean;
    hdl_colesterol: boolean;
    ldl_colesterol: boolean;
    trigliceride_serice: boolean;
    glicemie: boolean;
    tgo: boolean;
    tgp: boolean;
    uree_serica: boolean;
    creatina_serica: boolean;
    potasiu_seric: boolean;
    magneziu_seric: boolean;
    acid_uric: boolean;
    calciu_ionic_seric: boolean;
    calciu_seric_total: boolean;
    inr_cu_interpretare: boolean;
    hemoleucograma_completa: boolean;
    t3: boolean;
    t4: boolean;
    tsh: boolean;

    patient: Patient;
    
    constructor(){}
}
